ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -144,6 +173,9 @@ public void start() throws StoreException {
             new BlobStoreStats(storeId, index, config.storeStatsBucketCount, bucketSpanInMs, logSegmentForecastOffsetMs,
                 queueProcessingPeriodInMs, config.storeStatsWaitTimeoutInSecs, time, longLivedTaskScheduler,
                 taskScheduler, diskIOScheduler, metrics);
+        if (!storeDescriptor.alreadyExisted()) {
```

Comment:
do we need this qualifier anymore? I think `checkCapacityAndUpdateHelix()` is doing the right thing now (not sending helix messages if not required). So maybe we don't need this.

Also, I just thought of another scenario where we might want to do this check. Imagine we set the threshold to be 90 - partitions get marked RO. We decide that 90 is too conservative and update to 99. We want the stores to be marked RW on restart.

Reply:
Took out line and related code